emacsGcc deffered compliation fails

First of all thanks for packaging emacsGcc!

Now that libgccjit is in unstable I was able to compile emacsGcc just find and so far it's been working well however emacs crashed when I tried setting `(setq comp-deferred-compilation t)` first thing in my .emacs. It looks like the immediate cause is due to all the resulting `*.eln` files being empty. When I try compiling any file without loading it using `native-compile-async` I see the following error for every file compiled:
```
aarch64-unknown-linux-gnu-gcc-9.3.0: fatal error: cannot execute ‘as’: execvp: No such file or directory compilation terminated.
```
I tried adding binutils to my environment but than that leads to errors like:
```
/nix/store/bmjqkgnj34c0k1bsvralz0dr6xvyfgzb-binutils-2.31.1/bin/ld: cannot find crti.o: No such file or directory
/nix/store/bmjqkgnj34c0k1bsvralz0dr6xvyfgzb-binutils-2.31.1/bin/ld: cannot find crtbeginS.o: No such file or directory
/nix/store/bmjqkgnj34c0k1bsvralz0dr6xvyfgzb-binutils-2.31.1/bin/ld: cannot find -lgcc
Native elisp load failed: "/home/akyle/.emacs.d/straight/build/magit/eln-aarch64-unknown-linux-gnu-63761565f6c9e3bd/magit.eln", "/home/akyle/.emacs.d/straight/build/magit/eln-aarch64-unknown-linux-gnu-63761565f6c9e3bd/magit.eln: file too short"
Fatal error 11: Segmentation fault
Backtrace:
[ommitted as it's just memory addresses]
```
I tried adding libgccjit to my environment but that gave me the error:
```
error: getting status of '/nix/store/kq3hwnp0bdia9ld9445fy829laf46vaf-libgccjit-9.3.0/lib/lib': Too many levels of symbolic links
```

--- a/default.nix
+++ b/default.nix
@@ -19,7 +19,7 @@ let
     };
   });
 
-  mkGitEmacs = namePrefix: jsonFile: attrsFn:
+  mkGitEmacs = namePrefix: jsonFile:
     (self.emacs.override { srcRepo = true; }).overrideAttrs(old: (let
       repoMeta = super.lib.importJSON jsonFile;
       attrs = {
@@ -30,35 +30,23 @@ let
           repo = "emacs";
           inherit (repoMeta) sha256 rev;
         };
-        buildInputs = old.buildInputs ++ [ super.jansson super.harfbuzz.dev ];
         patches = [
           ./patches/tramp-detect-wrapped-gvfsd.patch
           ./patches/clean-env.patch
         ];
-        postPatch = ''
+        postPatch = old.postPatch + ''
           substituteInPlace lisp/loadup.el \
           --replace '(emacs-repository-get-version)' '"${repoMeta.rev}"' \
           --replace '(emacs-repository-get-branch)' '"master"'
         '';
       };
-    in attrs // attrsFn (old // attrs)));
+    in attrs));
 
-  emacsGit = mkGitEmacs "emacs-git" ./repos/emacs/emacs-master.json (_: { });
+  emacsGit = mkGitEmacs "emacs-git" ./repos/emacs/emacs-master.json;
 
-  emacsGcc = mkGitEmacs "emacs-gcc" ./repos/emacs/emacs-feature_native-comp.json (old: {
-    # When this is enabled, emacs does native compilation lazily after starting
-    # up, resulting in quicker package builds up-front, at the cost of slower
-    # running emacs until everything has been compiled. Since the elpa files in
-    # the nix store are read-only and we have binary caches, we prefer the
-    # longer AOT compilation instead of this flag.
-    # makeFlags = [ "NATIVE_FAST_BOOT=1" ];
-
-    LIBRARY_PATH = "${super.lib.getLib self.stdenv.cc.libc}/lib";
-
-    configureFlags = old.configureFlags ++ [ "--with-nativecomp" ];
-
-    buildInputs = old.buildInputs ++ [ self.libgccjit ];
-  });
+  emacsGcc = (mkGitEmacs "emacs-gcc" ./repos/emacs/emacs-feature_native-comp.json).override {
+    nativeComp = true;
+  };
 
   emacsUnstable = let
     repoMeta = super.lib.importJSON ./repos/emacs/emacs-unstable.json;
@@ -70,7 +58,6 @@ let
       repo = "emacs";
       inherit (repoMeta) sha256 rev;
     };
-    buildInputs = old.buildInputs ++ [ super.jansson super.harfbuzz.dev ];
     patches = [
       ./patches/tramp-detect-wrapped-gvfsd-27.patch
       ./patches/clean-env.patch

